src/nvim/tui/tui.c

```diff
@@ -1305,18 +1296,30 @@ void tui_default_colors_set(TUIData *tui, Integer rgb_fg, Integer rgb_bg, Intege
   invalidate(tui, 0, tui->grid.height, 0, tui->grid.width);
 }
 
-/// Enable synchronized output. When enabled, the terminal emulator will preserve the last rendered
-/// state on subsequent re-renders. It will continue to process incoming events. When synchronized
-/// mode is disabled again the emulator renders using the most recent state. This avoids tearing
-/// when the terminal updates the screen faster than Nvim can redraw it.
-static void tui_sync_output(TUIData *tui, bool enable)
+/// Begin flushing the TUI. If 'termsync' is set and the terminal supports synchronized updates,
+/// begin a synchronized update. Otherwise, hide the cursor to avoid cursor jumping.
+static void tui_flush_start(TUIData *tui)
+  FUNC_ATTR_NONNULL_ALL
 {
-  if (!tui->sync_output) {
-    return;
+  if (tui->sync_output && tui->unibi_ext.sync != -1) {
+    UNIBI_SET_NUM_VAR(tui->params[0], 1);
+    unibi_out_ext(tui, tui->unibi_ext.sync);
+  } else {
+    unibi_out(tui, unibi_cursor_invisible);
   }
+}
 
-  UNIBI_SET_NUM_VAR(tui->params[0], enable ? 1 : 0);
-  unibi_out_ext(tui, tui->unibi_ext.sync);
+/// Finish flushing the TUI. If 'termsync' is set and the terminal supports synchronized updates,
+/// end a synchronized update. Otherwise, make the cursor visible again.
+static void tui_flush_end(TUIData *tui)
+  FUNC_ATTR_NONNULL_ALL
+{
+  if (tui->sync_output && tui->unibi_ext.sync != -1) {
+    UNIBI_SET_NUM_VAR(tui->params[0], 0);
+    unibi_out_ext(tui, tui->unibi_ext.sync);
+  } else if (!tui->want_invisible) {
```

Comment:
Do we need to also check `tui->busy` in this condition?

Reply:
Yes. Yes we do.

Reply:
@jdebp @teto